src/Control/Effect/Exception.hs

```diff
@@ -0,0 +1,254 @@
+{-# LANGUAGE DeriveFunctor, ExistentialQuantification, RankNTypes, StandaloneDeriving #-}
+-- | Operations from "Control.Exception" lifted into effectful contexts using 'Control.Effect.Lift.Lift'.
+--
+-- @since 1.0.0.0
```

Comment:
This module is orthogonal to the other changes in this PR, and could be pulled out into a separate PR or into `fused-effects-exceptions` for that matter, but I thought it was worth exploring how to lift operations like `mask`, and then I kind of got carried away.

This module is exhaustive, in the sense that it lifts everything from `Control.Exception` which involves `IO` (most functions, + `Handler` for use with `catches`), and re-exports everything else.

Reply:
I actually kind of wonder if we shouldn’t move this to `fused-effects-exceptions` just because of the known issues with e.g. `finally` dropping state changes in the cleanup action when used with `MonadBaseControl`—those sorts of things absolutely apply to us as well.